### DISCUSSION TITLE
How to pass socket to java thread

### DISCUSSION THREAD

**Discussion by coderczp** [libuv/libuv#3427] [General]
Thank you very much for providing such a great open source library. I use the java version of libuv to implement a service similar to nginx(On my 4-core 8G computer, libuv version qps is 3000+, netty version is 2400+, nginx 2800+[Test with wrk]) but it can only use a single thread. How can I implement the master-woker mode? I am good at java code, but not I'm good at C code, I've been stuck on this problem for a week, I can't achieve it after reading a lot of information, please help me

libuv-java: https://github.com/webfolderio/libuv-java

**Comment by bnoordhuis** [comment]
You can `uv_write2()` them over a `uv_pipe_t` that's been opened in IPC mode with `uv_pipe_init(pipe, 1)`.

It's a little clunky but, on the positive side, it also works in a multi-process setup.

**Reply by coderczp** [reply]
Thanks for the reply, I will try your suggestion
